bigframes/streaming/__init__.py

```diff
@@ -13,17 +13,254 @@
+        on creating a pubsub topic, see
+        https://cloud.google.com/pubsub/docs/samples/pubsub-quickstart-create-topic?hl=en
+
+        Note that a service account is a requirement for continuous queries
+        exporting to pubsub.
+
+        Args:
+            topic (str):
+                The name of the pubsub topic to export to.
+                For example: "taxi-rides"
+            service_account_email (str):
+                Full name of the service account to run the continuous query.
+                Example: accountname@projectname.gserviceaccounts.com
+            job_id (str, default None):
+                If specified, replace the default job id for the query,
+                see job_id parameter of
+                https://cloud.google.com/python/docs/reference/bigquery/latest/google.cloud.bigquery.client.Client#google_cloud_bigquery_client_Client_query
+            job_id_prefix (str, default None):
+                If specified, a job id prefix for the query, see
+                job_id_prefix parameter of
+                https://cloud.google.com/python/docs/reference/bigquery/latest/google.cloud.bigquery.client.Client#google_cloud_bigquery_client_Client_query
+
+        Returns:
+            google.cloud.bigquery.QueryJob:
+                See https://cloud.google.com/python/docs/reference/bigquery/latest/google.cloud.bigquery.job.QueryJob
+                The ongoing query job can be managed using this object.
+                For example, the job can be cancelled or its error status
+                can be examined.
+        """
+        return _to_pubsub(
+            self.sql,
+            topic=topic,
+            service_account_email=service_account_email,
+            session=self._session,
+            job_id=job_id,
+            job_id_prefix=job_id_prefix,
+        )
+
+
+class StreamingDataFrame(StreamingBase):
+    __doc__ = _curate_df_doc(dataframe.DataFrame.__doc__)
+
+    # Private constructor
+    _create_key = object()
+
+    def __init__(self, df: dataframe.DataFrame, *, create_key=0):
+        if create_key is not StreamingDataFrame._create_key:
+            raise ValueError(
+                "StreamingDataFrame class shouldn't be created through constructor. Call bigframes.Session.read_gbq_table_streaming method to create."
+            )
+        self._df = df
+        self._df._disable_cache_override = True
+
+    @classmethod
+    def _from_table_df(cls, df: dataframe.DataFrame) -> StreamingDataFrame:
+        return cls(df, create_key=cls._create_key)
+
+    def __getitem__(self, *args, **kwargs):
+        return _return_type_wrapper(self._df.__getitem__, StreamingDataFrame)(
+            *args, **kwargs
+        )
+
+    __getitem__.__doc__ = _curate_df_doc(
+        inspect.getdoc(dataframe.DataFrame.__getitem__)
+    )
+
+    def __setitem__(self, *args, **kwargs):
+        return _return_type_wrapper(self._df.__setitem__, StreamingDataFrame)(
+            *args, **kwargs
+        )
```

Comment:
This could cause joins, which are not allowed. Should catch before query-time

Reply:
Yes, if the input is another Series, and if it is from another table, it's a join. And it will return an error for no index, I think? So it won't be sent to query?

Maybe it is a place that we need different definition and implementations with normal DF. Logged in b/356201125.